python/ray/rllib/models/extra_spaces.py

```diff
@@ -0,0 +1,58 @@
+import numpy as np
+from gym.spaces import Space
+
+
+class Simplex(Space):
+    """Represents a d - 1 dimensional Simplex in R^d.
+
+    That is, all coordinates are in [0, 1] and sum to 1.
+    The dimension d of the simplex is assumed to be shape[-1].
+
+    Additionally one can specify the underlying distribution of
+    the simplex as a Dirichlet distribution by providing concentration
+    parameters. By default, sampling is uniform, i.e. concentration is
+    all 1s.
+
+    Example usage:
+    self.action_space = spaces.Simplex(shape=(3, 4))
+        --> 3 independent 4d Dirichlet with uniform concentration
+    """
+
+    def __init__(self, shape, concentration=None, dtype=np.float32):
+        assert type(shape) in [tuple, list]
+        self.shape = shape
+        self.dtype = dtype
+        self.dim = shape[-1]
+
+        if concentration is not None:
+            assert concentration.shape == shape[:-1]
+        else:
+            self.concentration = [1] * self.dim
+
+        super().__init__(shape, dtype)
+        self.np_random = np.random.RandomState()
+
+    def seed(self, seed):
+        self.np_random.seed(seed)
+
+    def sample(self):
+        return np.random.dirichlet(
+            self.concentration, size=self.shape[:-1]).astype(self.dtype)
+
+    def contains(self, x):
+        assert x.shape == self.shape, (x.shape, self.shape)
+        assert np.allclose(np.sum(x, axis=-1), np.ones_like(x[..., 0]))
```

Comment:
Does it make sense to return False instead of asserting here?

Reply:
Yes it should actual return False when the sample is not in the given space. I've updated this part.